actors/evm/src/interpreter/precompiles/mod.rs

```diff
@@ -14,79 +14,117 @@ use fvm::{call_actor, get_actor_type, get_randomness, lookup_delegated_address,
 
 // really I'd want to have context as a type parameter, but since the table we generate must have the same types (or dyn) its messy
 type PrecompileFn<RT> = unsafe fn(*mut System<RT>, &[u8], PrecompileContext) -> PrecompileResult;
-pub type PrecompileResult = Result<Vec<u8>, PrecompileError>; // TODO i dont like vec
-
-/// Generates a list of precompile smart contracts, index + 1 is the address. (another option is to make an enum)
-const fn gen_precompiles<RT: Runtime>() -> [PrecompileFn<RT>; 14] {
-    macro_rules! precompiles {
-        ($($precompile:ident,)*) => {
-            mod trampolines {
-                use fil_actors_runtime::runtime::Runtime;
-                use crate::System;
-                use super::{PrecompileContext, PrecompileResult};
-                $(
-                    #[inline(always)]
-                    pub unsafe fn $precompile<RT: Runtime>(s: *mut System<RT>, inp: &[u8], ctx: PrecompileContext) -> PrecompileResult {
-                        super::$precompile(&mut *s, inp, ctx)
-                    }
-                )*
-            }
-            [
-                $(trampolines::$precompile,)*
-            ]
+pub type PrecompileResult = Result<Vec<u8>, PrecompileError>;
+
+pub const NATIVE_PRECOMPILE_ADDRESS_PREFIX: u8 = 0xFE;
+
+macro_rules! precompiles {
+    ($($precompile:ident,)*) => {
+        mod trampolines {
+            use fil_actors_runtime::runtime::Runtime;
+            use crate::System;
+            use super::{PrecompileContext, PrecompileResult};
+            $(
+                #[inline(always)]
+                pub unsafe fn $precompile<RT: Runtime>(s: *mut System<RT>, inp: &[u8], ctx: PrecompileContext) -> PrecompileResult {
+                    super::$precompile(&mut *s, inp, ctx)
+                }
+            )*
         }
+        [
+            $(trampolines::$precompile,)*
+        ]
+    }
+}
+
+/// Generates a list of precompile smart contracts, index + 1 is the address.
+const fn gen_evm_precompiles<RT: Runtime>() -> [PrecompileFn<RT>; 9] {
+    precompiles! {
+        ec_recover, // 0x01 ecrecover
+        sha256,     // 0x02 SHA2-256
+        ripemd160,  // 0x03 ripemd160
+        identity,   // 0x04 identity
+        modexp,     // 0x05 modexp
+        ec_add,     // 0x06 ecAdd
+        ec_mul,     // 0x07 ecMul
+        ec_pairing, // 0x08 ecPairing
+        blake2f,    // 0x09 blake2f
     }
+}
 
+const fn gen_native_precompiles<RT: Runtime>() -> [PrecompileFn<RT>; 5] {
+    // ? REMOVEME since we are already re-doing addressing, I'd like to put call_actor after lookup delegated
```

Comment:
please note

Reply:
re-order of the precompiles in the address space:
```
resolve_address,       
lookup_delegated_address,   
call_actor,                
get_actor_type,            
get_randomness, (remove this around the same time?)             
```

Reply:
> get_randomness, (remove this around the same time?)

Just for clean commits... I'd put this after call_actor, then remove it in the next PR (avoids reordering).